ReactAndroid/src/main/java/com/facebook/react/uimanager/UIManagerModule.java

```diff
@@ -85,6 +91,21 @@ public UIManagerModule(
     super(reactContext);
     mEventDispatcher = new EventDispatcher(reactContext);
     DisplayMetrics displayMetrics = reactContext.getResources().getDisplayMetrics();
+
+    // Get the real display metrics if we are using API level 17 or higher.
+    // The real metrics include system decor elements (e.g. soft menu bar).
+    //
+    // See: http://developer.android.com/reference/android/view/Display.html#getRealMetrics(android.util.DisplayMetrics)
+    try {
+      if (Build.VERSION.SDK_INT >= 17) {
```

Comment:
I would also be pro for adding the reflections-based code here for API16 so that we have better consistency across API versions: http://stackoverflow.com/a/23861333/573427. We should throw that exception too to know if it's not working. Please extract all this metrics determining logic into a separate method outside of the constructor though for readability.